### PR TITLE
added option to fail immediately after single test fails

### LINKED ISSUE #1373: Add option to fail immediately after single test fails
**Describe the feature**
Add option to fail immediately after single test fails, currently when running cargo tarpaulin if test fails during run only at the end it gives error, but all other tests are still runned
```
Error: "Test failed during run"
```

Add option to fail immediately after single test fails, this would make running faster, because all tests would not needed to be ran

### PR DESCRIPTION
**Pull request by 0xMimir**
When running code coverage on bigger projects, if test fails, tests for all other libraries and binaries are still runned, it would be nice to have a option to fail immediately after single test fails, like `cargo test` does.

Closes #1373

### CODE REVIEW

File: src/args.rs [hunk 1]
```diff
@@ -201,6 +201,9 @@ pub struct ConfigArgs {
     /// CI server being used, if unspecified tarpaulin may automatically infer for coveralls uploads
     #[arg(long, value_name = "SERVICE")]
     pub ciserver: Option<Ci>,
+    /// Option to fail immediately after single test fails
```
Comment (xd009642) [inline src/args.rs:204]: Grammar nit: "fail immediately after a single test fails"

File: src/main.rs [hunk 2]
```diff
@@ -24,7 +24,8 @@ fn print_env(seen_rustflags: HashMap<String, Vec<String>>, prefix: &str, default
 
 fn main() -> Result<(), String> {
     let args = CargoTarpaulinCli::from_args();
-
+    // println!("{:#?}", args.config);
```
Comment (xd009642) [inline src/main.rs:27]: Remove the commented out lines

**Review by xd009642 — CHANGES_REQUESTED** [review]
So two small tidiness comments, and if you can update the changelog something like as follows:

```
## [Unreleased]
### Added
- Added `--fail-immediately` flag to abort execution the moment the first test failure occurs

### Changed
```

**Comment by 0xMimir** [comment]
Fixed typos, removed commended lines and added changes to `CHANGELOG.md`

**Comment by xd009642** [comment]
Ah and cargo fmt is failing in CI so if you can run `cargo fmt` and push the resulting changes as well

**Comment by 0xMimir** [comment]
I have added changes from cargo fmt

**Pull request merged**